src/DDTrace/Encoders/Json.php

```diff
@@ -35,6 +39,7 @@ private function encodeSpan(Span $span)
     {
         $json = json_encode($this->spanToArray($span));
         if (false === $json) {
+            $this->logger->debug("Failed to json-encode span: " . json_last_error_msg());
```

Comment:
Is `$this->logger` always initialized at this point?

Reply:
Taken care of with constructor injection